src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedSearch.tsx

```diff
@@ -0,0 +1,168 @@
+      } else if (selectedClass?.id) {
+        searchParams.push(`class=${selectedClass.id}`);
+      }
+
+      reqParamObject.params = searchParams.join('&');
+      return reqParamObject;
+    },
+    [phaseProposal, phaseDesign, selectedSubClass, selectedClass],
+  );
+
+  const { t } = useTranslation();
+
+  const handleValueChange = useCallback(
+    (value: string) => setSearchState((current) => ({ ...current, searchWord: value })),
+    [],
+  );
+
+  const getSuggestions = useCallback(
+    (inputValue: string) => {
+      return new Promise<{ value: string; label: string }[]>((resolve, reject) => {
+        // printing out search params for later
+        const queryString = buildQueryParamString(inputValue);
+        console.log(queryString);
+
+        getProjectsWithParams(queryString)
+          .then((res) => {
+            // Filter out only the projects which haven't yet been added to be the submitted list from the result
+            const projectsIdList = projectsForSubmit.map((p) => p.value);
+            const resultList = res.results.filter(
+              (object) => object.type === 'projects' && !arrayHasValue(projectsIdList, object.id),
+            );
+
+            // Convert the resultList to options for the suggestion dropdown
+            const searchProjectsItemList: Array<IProgrammedProjectSuggestions> | [] = resultList
+              ? resultList.map((project) => ({
+                  label: project.name,
+                  value: project.id,
+                  breadCrumbs: buildBreadCrumbs(project.path, classes),
+                }))
+              : [];
+            if (searchProjectsItemList.length > 0) {
+              setSearchState((current) => ({
+                ...current,
+                searchedProjects: searchProjectsItemList,
+              }));
+            }
+
+            resolve(searchProjectsItemList);
+          })
+          .catch(() => reject([]));
+      });
+    },
+    [projectsForSubmit, buildQueryParamString, buildBreadCrumbs, classes],
+  );
+
+  const handleSubmit = useCallback(
+    (value: string) => {
+      const selectedProject = searchedProjects.find((p) => p.label === value);
+
+      if (value) {
+        onProjectClick(selectedProject);
+      }
+
+      setSearchState((current) => ({ ...current, searchWord: '' }));
+    },
+    [onProjectClick, searchedProjects],
+  );
+
+  const handleDelete = useCallback(
+    (name: string) => {
+      onProjectSelectionDelete(name);
+    },
+    [onProjectSelectionDelete, projectsForSubmit],
+  );
+  return (
+    <div className="project-search-input" data-testid="search-project-field-section">
+      <SearchInput
+        label={t('searchForProjects')}
+        getSuggestions={getSuggestions}
+        clearButtonAriaLabel="Clear search field"
+        searchButtonAriaLabel="Search"
+        suggestionLabelField="label"
+        value={searchWord}
+        onChange={handleValueChange}
+        onSubmit={(v) => handleSubmit(v)}
```

Comment:
remove anonymous function. give handleSubmit function as onSubmit value. Refactor from onSubmit={(v) => handleSubmit(v)} to onSubmit={handleSubmit}

Reply:
done